packages/platform-core/src/errors/amplify_error.ts

```diff
@@ -47,31 +47,49 @@ export abstract class AmplifyError<T extends string = string> extends Error {
     if (cause && cause instanceof AmplifyError) {
       cause.serializedError = undefined;
     }
-    this.serializedError = JSON.stringify({
-      name,
-      classification,
-      options,
-      cause,
-    });
+    this.serializedError = JSON.stringify(
+      {
+        name,
+        classification,
+        options,
+        cause,
+      },
+      errorSerializer
+    );
   }
 
   static fromStderr = (_stderr: string): AmplifyError | undefined => {
+    /**
+     * `["']?serializedError["']?:[ ]?` captures the start of the serialized error. The quotes depend on which OS is being used
+     * `(?:`(.+?)`|'(.+?)'|"((?:\\"|[^"])*?)")` captures the rest of the serialized string enclosed in either single quote,
+     * double quotes or back-ticks.
+     */
```

Comment:
Thanks for adding this, the regex is definitely hard to read so this helps in understanding what's going on.